### PR TITLE
[#19] Update Encoder implementation.

### PR DESCRIPTION
**Pull request by jaeheungs**
#19 

-  Add `train_backbone` flag to Encoder initializer and model selector to freeze or unfreeze backbone weights
-  reorganized Encoder implementation files

### CODE REVIEW

File: image_captioning/models/InceptionV3Encoder.py [hunk 1]
```diff
@@ -8,24 +8,29 @@ class InceptionV3Encoder(tf.keras.Model):
...
         """
         super().__init__()
         self.embedding_dim = embedding_dim
 
         # Using Inception V3 backbone
         self.backbone = tf.keras.applications.InceptionV3(include_top=False,
                                                           weights='imagenet')
+        if not train_backbone:
+            for layer in self.backbone.layers:
+                layer.trainable = False
```
Comment (dansuh17) [inline image_captioning/models/InceptionV3Encoder.py]: Isn't `self.backbone.trainable = train_backbone` enough to set trainability? I was referring [this example](https://www.tensorflow.org/guide/keras/transfer_learning#the_typical_transfer-learning_workflow) and it seems it sets `trainable = False` only to the outer model.

**Review by dansuh17 — APPROVED** [review]
LGTM w/ minor comment.

File: image_captioning/models/InceptionV3Encoder.py (hunk 1 above)
Comment (jaeheungs) [reply image_captioning/models/InceptionV3Encoder.py]: lemme check

Comment (jaeheungs) [reply image_captioning/models/InceptionV3Encoder.py]: I think that's right. Lemme change it for simplicity's sake.

**Pull request merged**